crates/sui-framework/packages/dwallet-system/sources/eth_dwallet.move

```diff
@@ -39,8 +44,22 @@ module dwallet_system::eth_dwallet {
         dwallet_id: vector<u8>,
```

Comment:
```suggestion
```
Take this from the `EthereumDWalletCap`

Reply:
It is not there.
inside `EthereumDWalletCap` we only have the `DWalletCap` ID.

Reply:
So get `DWallet`, and check that it matches the ID of the `EthereumDWalletCap.dwallet_cap`. But you must verify the user-passed argument `dwallet_id` somehow, or better even, not get it from the user and extract from a Move struct.